apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -6085,27 +6093,31 @@ sc_ws_broken_call_code_(Owner, GetVolley, _CreateContract, _ConnPid1, _ConnPid2,
     #{<<"return_type">> := <<"error">>} = Res,
     ok.
 
-mine_micro_block_emptying_mempool_or_fail(Node) ->
-    {ok, [KeyBlock, MicroBlock]} = aecore_suite_utils:mine_micro_blocks(Node, 1),
+mine_micro_block_emptying_mempool_or_fail(Node, 0) ->
+    {error, retries_exhausted};
+mine_micro_block_emptying_mempool_or_fail(Node, Retries) ->
+    %% Very rarely we get two key-blocks followed by a microblock
+    {ok, Blocks} = aecore_suite_utils:mine_micro_blocks(Node, 1),
+    [MicroBlock, KeyBlock | _] = lists:reverse(Blocks),
     true = aec_blocks:is_key_block(KeyBlock),
     false = aec_blocks:is_key_block(MicroBlock),
+
     case rpc:call(Node, aec_tx_pool, peek, [infinity]) of
-        {ok, []} -> ok;
+        {ok, []} ->
+            {ok, [KeyBlock, MicroBlock]};
         {ok, [_|_] = MempoolTxs} ->
+            %% So, Tx(s) is/are back in the mempool, this means (unless some Txs arrived
+            %% from thin air) that we had a micro-fork. Let's check what state we stopped in
+            timer:sleep(100), %% and avoid races...
             NewBlocks = aecore_suite_utils:flush_new_blocks(),
-            case
-                (length(NewBlocks) =:= 1) andalso aec_blocks:is_key_block(hd(NewBlocks)) andalso
-                (aec_blocks:hash_internal_representation(KeyBlock) =:= {ok, aec_blocks:prev_key_hash(MicroBlock)}) andalso
-                (aec_blocks:prev_key_hash(MicroBlock) =:= aec_blocks:prev_key_hash(hd(NewBlocks)))
-            of
-                true ->
-                    ct:fail("Key block mined shortly after micro block mined pushed the micro block out of the chain.");
-                false ->
-                    ct:pal(
-                        "Unexpected transactions in mempool:~n~p~nUnexpected new block events:~n~p",
-                        [MempoolTxs, NewBlocks]),
-                    ct:fail("Unexpected transactions in mempool")
+            case [ aec_blocks:type(B) || B <- NewBlocks ] of
+                [key, micro] -> %% We had a microfork but then added the Tx again!
+                    {ok, NewBlocks};
+                [key]        -> %% We had a microfork and stopped, we can retry
+                    mine_micro_block_emptying_mempool_or_fail(Node, Retries - 1);
+                _Other       ->
```

Comment:
Would it make sense to also here continue mining, until for example `length(NewBlocks)`> 5... that is, give it some more change to succeed...

and then really fail if that does not work? Not sure here.